compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -1745,6 +1745,9 @@ pub struct ParamEnv<'tcx> {
     ///
     /// Note: This is packed, use the reveal() method to access it.
     packed: CopyTaggedPtr<&'tcx List<Predicate<'tcx>>, traits::Reveal, true>,
+
+    /// FIXME: This field is not used, but removing it causes a performance degradation. See #76913.
+    pub unused_field: Option<DefId>,
```

Comment:
I would like this to be made private.

Maybe we should also replace `Option<DefId>` here with something like `MaybeUninit<u64>`, which would never be initialized, and see if that's still enough to avoid the regression?

Reply:
Private: good point!

About the different type: I'm not sure what a nice solution would be. `MaybeUninit` is not `Eq`, for example. I now wrapped an array of `u8`'s to stress that the side (probably) should be 8 bytes. What do you think?